modules/phase_field/include/kernels/KKSMultiPhaseConcentration.h

```diff
@@ -9,7 +9,8 @@
 
 #pragma once
 
-#include "Kernel.h"
+#include "KernelValue.h"
```

Comment:
What is the advantage/upside for changing to deriving from `KernelValue` rather than `Kernel`? To be honest, it seems like for a newer user, the changes take them further from what they learn in the MOOSE training and increases the learning curve (unless that aspect of MOOSE training has changed). Granted this is not a kernel that a brand new user is likely going to dig into as they are getting started, so I'm not saying I'm necessarily opposed to it, I just want to make sure there is a good reason for the added complexity.

Reply:
The advantage is a reduced number of residual evaluations. We should really always be using `KernelValue` for kernels with residuals of the form (..., test) and `KernelGrad` for kernels with residuals of the form (..., grad test). 
IMO this is a design decision that has been made long ago and we should stick with it.